javascript/ql/src/semmle/javascript/dataflow/internal/BasicExprTypeInference.qll

```diff
@@ -92,10 +92,35 @@ private class AnalyzedArrayComprehensionExpr extends DataFlow::AnalyzedValueNode
 /**
  * Flow analysis for functions.
  */
-private class AnalyzedFunction extends DataFlow::AnalyzedValueNode {
+class AnalyzedFunction extends DataFlow::AnalyzedValueNode {
   override Function astNode;
 
   override AbstractValue getALocalValue() { result = TAbstractFunction(astNode) }
+
+  /**
+   * Gets a return value for this function.
+   * 
+   * This does not model the case where the function is invoked as a constructor.
```

Comment:
Maybe just add that to the first sentence? "Gets a return value for this function when invoked without `new`" or something like that.